feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts

```diff
@@ -278,6 +278,17 @@ export class ConfiguratorAddToCartButtonComponent implements OnInit, OnDestroy {
       );
   }
 
+  extractConfigPrices(configuration: Configurator.Configuration) {
+    let priceSummary = configuration?.priceSummary;
+    if (priceSummary) {
+      return {
+        basePrice: priceSummary?.basePrice?.formattedValue,
+        selectedOptions: priceSummary?.selectedOptions?.formattedValue,
+        totalPrice: priceSummary?.currentTotal?.formattedValue,
+      };
```

Comment:
Because you put an if in line 283, the optional chaining operators (behind price summary)  in line 285, 286 and 287 are superfluous. However I'm not sure what happens if you really intend to return `undefined`in case configuration and/or priceSummary is not defined?